Microsoft.Toolkit.Uwp.UI.Animations/CompositionAnimations/Animations/AnimationBase.cs

```diff
@@ -0,0 +1,147 @@
+            set { SetValue(DurationProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="KeyFrameCollection"/> of the animations
+        /// </summary>
+        public KeyFrameCollection KeyFrames
+        {
+            get
+            {
+                var collection = (KeyFrameCollection)GetValue(KeyFramesProperty);
+                if (collection == null)
+                {
+                    collection = new KeyFrameCollection();
+                    SetValue(KeyFramesProperty, collection);
+                }
+
+                return collection;
+            }
+
+            set
+            {
+                SetValue(KeyFramesProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the target property to be animated
+        /// </summary>
+        public string Target
+        {
+            get { return (string)GetValue(TargetProperty); }
+            set { SetValue(TargetProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the property that should start the implicit animation
+        /// </summary>
+        public string ImplicitTarget
+        {
+            get { return (string)GetValue(ImplicitTargetProperty); }
+            set { SetValue(ImplicitTargetProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the delay of the animation
+        /// </summary>
+        public TimeSpan Delay
+        {
+            get { return (TimeSpan)GetValue(DelayProperty); }
+            set { SetValue(DelayProperty, value); }
+        }
+
+        private static bool? _isCreatorsUpdateOrAbove;
+
+        /// <summary>
+        /// Gets a value indicating whether Universal API Contract 4 is available on the current device
+        /// where Implicit Show/Hide animations and Translation property on <see cref="Visual"/> are available
+        /// </summary>
+        public static bool IsCreatorsUpdateOrAbove => (bool)(_isCreatorsUpdateOrAbove ??
+            (_isCreatorsUpdateOrAbove = ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 4)));
+
+        /// <summary>
+        /// Gets a <see cref="CompositionAnimation"/> that can be used on the Composition layer
+        /// </summary>
+        /// <param name="compositor">The <see cref="Compositor"/> to use to create the animation</param>
+        /// <returns><see cref="CompositionAnimation"/></returns>
+        public abstract CompositionAnimation GetCompositionAnimation(Compositor compositor);
+
+        /// <summary>
+        /// Called when any property of the animation changes
+        /// </summary>
+        /// <param name="d">The animation where a property has changed</param>
+        /// <param name="e">The details about the property change</param>
+        protected static void OnAnimationPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
```

Comment:
A protected static method seems odd. Why not have an instance method that's called that you would be overriding?Also you'd be overriding one method that's called for multiple different dependency properties. It would be cleaner to have one method per property.

Reply:
Why one method per property if the action is not specific to a property and can all be done in a single method? It's not cleaner and seems more error prone.

Reply:
Done